airbyte-lib/airbyte_lib/caches/base.py

```diff
@@ -415,10 +441,15 @@ def _create_table(
         self,
         table_name: str,
         column_definition_str: str,
+        primary_keys: list[str] | None = None,
```

Comment:
Primary keys are not set on the final table.

I think it's because this argument is never passed in, I think it should be on line 404 where the final table is created (`self._create_table(table_name, column_definition_str, self._get_primary_keys(stream_name))`)

not sure whether there is an advantage of setting the primary key for the loading tables (I would guess not, maybe the join that's required for the merge would be slightly faster but OTOH the temporary tables are pretty small so that shouldn't change a thing).